### ISSUE TITLE
[3.0] CurveTextureEditor allows making curls: bug or feature?

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#8504]
Godot master

I gave a try at the CurveTextureEditor, and I found it allowed making loops, or "overhanging" parts, which is normally only allowed in actual 2D curves, which this one isn't because it is supposed to save to a texture with only one row of pixel values. But with such loops you would end up with two Y values to choose for one X value.

![image](https://cloud.githubusercontent.com/assets/1311555/25318356/44080ef2-288d-11e7-83a7-9138d140259d.png)
Is it a bug or a feature?

Note: I was considering to re-use this editor, as I started to work on a Curve resource for https://github.com/godotengine/godot/issues/7296
however I might end up not to, because features between the two differ.

**Comment by n-pigeon** [comment]
I think it's a bug. It probably uses same curve widget that visual programming and both should be fixed :D

**Comment by reduz** [comment]
Needs to be fixed or replaced by actual splines i think

**Comment by Zylann** [comment]
Could be fixed with https://github.com/godotengine/godot/issues/7296#issuecomment-297201453

**Comment by Zylann** [comment]
I was thinking about removing CurveTextureEditorPlugin, and simply have a Curve property in CurveTexture, so it can have a broader use. What do you think?

**Comment by reduz** [comment]
Nope.. too much info for a single property

**Comment by Zylann** [comment]
Reusing the same editor would be handy, though. Otherwise it will be copy/paste...

**Comment by reduz** [comment]
Yeah, that definitely.. I haven't changed it much because i want to make it
actual splines

**Comment by Zylann** [comment]
I almost finished one for a Curve resource (se other resource), works similarly to Unity

**Comment by reduz** [comment]
If you want to modify it for curve texture, that would be great

**Comment by Zylann** [comment]
The main difference between the two is that I made the Curve editor work like with other curves, it doesn't bakes it, the resource does (so it works at runtime too). As a result, it also takes a Curve reference rather than its point data or an interface.
It could work with CurveTexture by having an internal Curve which is then used to bake the CurveTexture, which is why I proposed the curve property, but I agree the "two-step" workflow is not ideal.
This is mostly implementation details, I'm not decided yet on how it will be re-used.
